frame/contracts/primitives/src/lib.rs

```diff
@@ -71,15 +71,26 @@ pub struct ContractResult<R, Balance> {
 	pub debug_message: Vec<u8>,
 	/// The execution result of the wasm code.
 	pub result: R,
+	/// The events that were emitted during execution. It is an options as event collection is
+	/// optional.
+	pub events: Option<Vec<EventRecord>>,
```

Comment:
This has been tested against Contracts UI and it works just fine.
The 'old' `call` and `instantiate` functions return `events: None` that encodes to `0x00`, but as far as I understand SCALE ignores trailing data. Anyway, if we want to be on the safe side we can always create two versions of `ContractResult`, one with and another one without `events`.
WDYT?

Reply:
> but as far as I understand SCALE ignores trailing data

SCALE is not really well defined by a spec or something. Whether you ignore trailing data or not is up to the implementation. See [`Decode`](https://docs.rs/parity-scale-codec/3.4.0/parity_scale_codec/trait.Decode.html) vs [`DecodeAll`](https://docs.rs/parity-scale-codec/3.4.0/parity_scale_codec/trait.DecodeAll.html#tymethod.decode_all).

That said, I suspect the default behavior is to ignore trailing data. Hence we can safely extend the struct in new versions. But if you think about it this means we don't even need a new version in this case because we can just always emit the events since they will be ignored by older code as trailing data. If they ignore a single `0x00` they will also ignore everything else.

Reply:
I have removed the new version and implemented it all in the original `ContractsApi`. Tested it against Contracts UI and works fine

Reply:
Can you add a `#Note` to the type to warn users that we take the liberty to extend the type with new data without bumping the API version? It should warn them to always ignore trailing data.